include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -1004,8 +1012,8 @@ void mergeSubproblems(SizeType i_begin, SizeType i_split, SizeType i_end, WorkSp
   // - reorder columns in `mat_ev` using `i2` such that eigenvectors match eigenvalues
   //
   sortIndex(i_begin, i_end, k_fut, ws.dtmp, ws.i1, ws.i2);
-  applyIndex(i_begin, i_end, ws.i2, ws.dtmp, d);
-  dlaf::permutations::permute<Backend::MC, Device::CPU, T, Coord::Col>(i_begin, i_end, ws.i2, ws.mat1,
-                                                                       mat_ev);
+  applyIndex(i_begin, i_end, ws.i2, ws.dtmp, ws_h.evals.get());
+  dlaf::permutations::permute<backend, device, T, Coord::Col>(i_begin, i_end, ws.i2, ws.mat1, evecs);
+  ws_h.evecs.copySourceToTarget();  // copy from GPU to CPU if evecs in on GPU
```

Comment:
Is this required because subsequent iterations may access target-side values that this iteration wrote into the source-side of `evecs`?

Reply:
Yes, exactly. This is a temporary situation as more kernels are moved to the GPU.